docs/index-server.md

```diff
@@ -0,0 +1,204 @@
+
+##### spark-defaults.conf(only for secure mode)
+
+| Name     |      Default Value    |  Description |
+|:----------:|:-------------:|:------:       |
+| spark.carbon.indexserver.principal |  NA | Used for authentication, whether a valid application is trying to connect to the server or not. Set in both IndexServer and JDBCServer     |
+| spark.carbon.indexserver.keytab |    NA   |   Specify the path to the keytab file through which authentication would happen. Set in both IndexServer and JDBCServer| 
+| spark.dynamicAllocation.enabled | true | Set to false, so that spark doesn’t kill the executor, If executors are killed, cache would be lost. Applicable only for Index Server. |
+| spark.yarn.principal | NA | Should be set to the same user used for JDBCServer. Required only for IndexServer   |
+|spark.yarn.keytab| NA | Should be set to the same as JDBCServer.   |
+
+##### spark-defaults.conf(non-secure mode)
+| Name     |      Default Value    |  Description |
+|:----------:|:-------------:|:------:       |
+| spark.dynamicAllocation.enabled | true | Set to false, so that spark doesn’t kill the executor, If executors are killed, cache would be lost. Applicable only for Index Server. |
+
+
+**NOTE:** Its better to create a new user for indexserver principal,
+that will authenticate the user to access the index server and no other
+application.
+
+##### core-site.xml
+
+| Name     |      Default Value    |  Description |
+|:----------:|:-------------:|:------:       |
+| ipc.client.rpc-timeout.ms |  NA | Set the above property to some appropriate value based on your estimated query time. The best option is to set this to the same value as spark.network.timeout|
+
+##### dynamic-properties(set command)
+
+| Name     |      Default Value    |  Description |
+|:----------:|:-------------:|:------:       |
+| carbon.enable.index.server |  false | Enable the use of index server for pruning for the current session |
+| carbon.enable.index.server.dbName.tableName |  false | Enable the use of index server for the specified table in the current session |
+  
+  
+## Starting the Server
+``` 
+./bin/spark-submit --master [yarn/local] --[optional parameters] --class org.apache.carbondata.indexserver.IndexServer [path to carbondata-spark2-<version>.jar]
+```
+Or 
+``` 
+./sbin/start-indexserver.sh --master yarn --num-executors 2 /<absolute path>/carbondata-spark2-1.6.0.0100.jar
+```
+
+
+## FAQ
+
+Q. **Index Server is throwing Large response size exception.** 
+
+A. The 
+exception would show the size of response it is trying to send overthe
+network. Use ipc.maximum.response.length to a value bigger than the
+response size.
+
+Q. **Index server is throwing Kerberos principal not set exception**
+
+A. set spark.carbon.indexserver.principal to the correct principal in both IndexServer and JDBCServer configurations.
```

Comment:
All the sentence should start from caps
set spark.carbon.ind.....   > Set spark.carbon.ind....

Reply:
done